### PR TITLE
[TAS-2030] 💄 Add inLanguage field for book

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-08-27 下午2 48 49

### CODE REVIEW

File: components/IscnRegisterForm.vue [hunk 1]
```diff
@@ -850,6 +866,10 @@ export default class IscnRegisterForm extends Vue {
     return 'CreativeWork'
   }
 
+  get defaultLanguage() {
+    return this.epubMetadata?.language || 'zh'
+  }
```
Comment (williamchong) [inline components/IscnRegisterForm.vue:871]: Should set `this.language` as `defaultLanguage` on `mounted()`

Comment (williamchong) [reply components/IscnRegisterForm.vue:871]: Also would be nice if we can set default as `en` or `zh` according to if we have non-ascii charater in `title` or `description`

**Review by williamchong — CHANGES_REQUESTED** [review]

File: components/IscnRegisterForm.vue (hunk 1 above)
Comment (AuroraHuang22) [reply components/IscnRegisterForm.vue]: Use @watch to detect name & description language for book.

[🚸 Auto detect language for book](https://github.com/likecoin/app-like-co/pull/471/commits/6af5f50ddf8443855607eca376d6ba4e8c009f99)

File: components/IscnRegisterForm.vue [hunk 2]
```diff
@@ -850,6 +866,17 @@ export default class IscnRegisterForm extends Vue {
     return 'CreativeWork'
   }
 
+  get defaultLanguage() {
+    const containsChinese = (str: any) => /[\u4E00-\u9FFF\u3400-\u4DBF\uF900-\uFAFF]/.test(str);
```
Comment (williamchong) [inline components/IscnRegisterForm.vue:870]: Suggest to just simply check for all ascii/alphanumeric for English and avoid unicode logic

File: components/IscnRegisterForm.vue [hunk 3]
```diff
@@ -850,6 +866,17 @@ export default class IscnRegisterForm extends Vue {
...
+
+    const isEpubInChinese = this.epubMetadata?.language === 'zh'
+
+    const titleContainsChinese = this.name ? containsChinese(this.name) : isEpubInChinese;
+    const descriptionContainsChinese = this.description ? containsChinese(this.description) : isEpubInChinese;
+
+    return (titleContainsChinese || descriptionContainsChinese) ? 'zh' : 'en';
+  }
```
Comment (williamchong) [inline components/IscnRegisterForm.vue:878]: Cleaner to just at the end `return (isEpubInChinese  || titleContainsChinese || descriptionContainsChinese) ? 'zh' : 'en'` and don't use `isEpubInChinese` in 874 and 875

**Review by williamchong — CHANGES_REQUESTED** [review]

File: components/IscnRegisterForm.vue (hunk 3 above)
Comment (AuroraHuang22) [reply components/IscnRegisterForm.vue]: [🎨 Simplify default language check](https://github.com/likecoin/app-like-co/pull/471/commits/7013a40c386ad473a0d62cd0b2a72b6d17512961)

File: components/IscnRegisterForm.vue [hunk 4]
```diff
@@ -850,6 +866,17 @@ export default class IscnRegisterForm extends Vue {
     return 'CreativeWork'
   }
 
+  get defaultLanguage() {
+    const containsOnlyASCII = (str: string) => /^[ -~]*$/.test(str);
+
+    const isEpubInChinese = this.epubMetadata?.language === 'zh';
```
Comment (williamchong) [inline components/IscnRegisterForm.vue:872]: Should we respect if epub metadata language is explicitly set to en?

Comment (AuroraHuang22) [reply components/IscnRegisterForm.vue]: [🎨 Prioritize language from epubMetadata](https://github.com/likecoin/app-like-co/pull/471/commits/7f5a5a736db1c90c64104b9d9de553a2262d92d4)

**Review by williamchong — APPROVED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**